open_spiel/examples/alpha_zero_torch_example.cc

```diff
@@ -116,43 +158,10 @@ int main(int argc, char** argv) {
     std::string config_string = config_file.ReadContents();
     open_spiel::json::Object config_json = open_spiel::json::FromString(
         config_string).value().GetObject();
-
-    config.FromJson(config_json);
+    config.FromJsonWithDefaults(config_json, config.ToJson());
```

Comment:
Correct.  Resuming from a model would continue with the saved values for all the settings specified in the file, and with the default values for settings created since the file was saved.  So a model saved before this PR would resume with defaults for `td_lambda` and `td_n_steps`. 

Adding a unit test for the config loading behavior should be easy, so I'll try to add that. And maybe too for the MCTS change.  For the other changes: unit tests for those are harder at the moment because there are no existing az tests to build off and the methods aren't even exposed in a header file.  For verifying the current `TdLambdaReturns` implementation: note I added logging and pasted some results for various settings in the PR description.  So you can see it's computing returns as expected at least for now.

Reply:
I added tests for the AlphaZeroConfig changes and the min_simulations change to MCTS.  But (sorry) tests for TdLambdaReturns inside alpha_zero_torch are left as a todo for later.  

Aside question:  It looks like the only difference between the C++ alpha_zero and alpha_zero_torch are the model source files, which already seem effectively abstracted via the Evaluator/VPNetEvaluator classes, so should be somewhat straightforward to unify the code, right?  If so, then a unifying refactor should help reduce the number of tests needed for coverage.

Reply:
Marking this resolved

Reply:
Heads up that in the short-term: we'll likely be depracating the TF-based C++ AlphaZero (which only works internally anyway_ but keeping the Python TF AlphaZero. Not sure how that affects this conevrsation, if at all, just letting you know.

Reply:
Ok, got it - makes sense and in that case I won't entertain thoughts of trying to unify the TF and torch code.  Late responses are always fine and appreciated - no need for hurry and I am using separate local branches for hacking anyway, so having open PRs is no matter to me.